services/profile.ts

```diff
@@ -0,0 +1,115 @@
+import { $fetch, FetchError } from 'ofetch'
+
+const hostToServiceMap = {
+  'kodadot.xyz': 'https://profile.kodadot.workers.dev/',
+  'canary.kodadot.xyz': 'https://profile-beta.kodadot.workers.dev/',
+  'localhost:9090': 'http://localhost:8787',
+}
+
+const BASE_URL = hostToServiceMap[window.location.host]
+
+const api = $fetch.create({
+  baseURL: BASE_URL,
+})
+
+// Types for API request and response objects
+export type Profile = {
+  address: string
+  name: string
+  description: string
+  image: string
+  banner: string
+  socials: SocialLink[]
+}
+
+export type Follower = {
+  address: string
+  name: string
+  description: string
+  image: string
+}
+
+export type SocialLink = {
+  handle?: string
+  platform: string
+  link: string
+}
+
+export type ProfileResponse = {
+  success: boolean
+  message: string
+  data?: Profile
+  profileId?: string
+}
+
+export type CreateProfileRequest = {
+  address: string
+  name: string
+  description: string
+  image: string
+  banner: string
+  socials: SocialLink[]
+}
+
+export type UpdateProfileRequest = {
+  name?: string
+  description?: string
+  image?: string
+  banner?: string
+}
+
+// API methods
+
+export const fetchProfiles = (limit?: number) =>
+  api<Profile[]>('/profiles', {
+    method: 'GET',
+    query: { limit },
+  })
+
+export const fetchProfileByAddress = (address: string) =>
+  api<Profile>(`/profiles/${formatAddress(address, 42)}`, {
+    method: 'GET',
+  })
+
+export const fetchFollowersOf = (address: string) =>
+  api<{ followers: Follower[] }>(
+    `/follow/${formatAddress(address, 42)}/followers`,
+    {
+      method: 'GET',
+    },
+  ).then((res) => res.followers)
+
+export const fetchFollowing = (address: string) =>
+  api<{ following: Follower[] }>(
```

Comment:
pagination for this API. It's good to have in case of huge followers data.  

It could be done on another issue.
```suggestion
  api<{ following: Follower[], total: number }>(
```

Reply:
i'll create a better API for this on the worker side